File: Offer2/0006/main.swift
public class ListNode {
    public var val: Int
    public var next: ListNode?
    public init() { self.val = 0; self.next = nil; }
    public init(_ val: Int) { self.val = val; self.next = nil; }
    public init(_ val: Int, _ next: ListNode?) { self.val = val; self.next = next; }
}

class Solution {
    func reversePrint(_ head: ListNode?) -> [Int] {
        var head = head
        var stack = [Int]()
        while head != nil{
            stack.append(head!.val)
            head = head!.next
        }
        var res = [Int]()
        while !stack.isEmpty{
            res.append(stack.removeLast())
        }
        return res
    }
}

File: Offer2/0006/main.py
from collections import deque
from typing import List


class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def reversePrint(self, head: ListNode) -> List[int]:
        stack: deque[int] = deque([])
        while head != None:
            stack.append(head.val)
            head = head.next
        res: List[int] = []
        while len(stack) > 0:
            res.append(stack.pop())
        return res